File: go/easy/Island Perimeter.go
package easy

func islandPerimeter(grid [][]int) int {
	perimetr := 0

	for i, row := range grid {
		for j, _ := range row {
			if grid[i][j] == 1 {
				perimetr += 4
				if i > 0 && grid[i-1][j] == 1 {
					perimetr -= 2
				}
				if j > 0 && grid[i][j-1] == 1 {
					perimetr -= 2
				}
			}
		}

	}

	return perimetr
}

File: python/easy/Island Perimeter.py
from typing import List


class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        perimetr = 0
        rows, cols = len(grid), len(grid[0])
        for i in range(rows):
            for j in range(cols):
                if grid[i][j]:
                    perimetr += 4
                    if i > 0 and grid[i-1][j]:
                        perimetr -= 2
                    if j > 0 and grid[i][j-1]:
                        perimetr -= 2


        return perimetr
